python/paddle/v2/__init__.py

```diff
@@ -85,22 +79,63 @@ def set_env(key, value):
         if envset is None:
             os.environ[key] = value
 
-    ht = os.popen("lscpu |grep \"per core\"|awk -F':' '{print $2}'|xargs")
-    ht = int(ht.read())
-    if ht == 1:  # ht is off
-        set_env("OMP_DYNAMIC", "false")
-        set_env("KMP_AFFINITY", "granularity=fine,compact,0,0")
-    else:
+    def is_hyperthreading_enabled():
+        '''If Hyper Threading is enabled.'''
+        if platform.system() == "Linux":
+            percore = os.popen(
+                "lscpu |grep \"per core\"|awk -F':' '{print $2}'|xargs")
+            return int(percore.read()) != 1
+        elif platform.system() == "Darwin":
+            physical = int(os.popen("sysctl hw.physicalcpu").read())
+            logical = int(os.popen("sysctl hw.logicalcpu").read())
+            return logical > physical
+        else:
+            # do not support on other platform yet
+            return False
+
+    def get_logical_processors():
```

Comment:
This function returns the number of logical processors, other than a list of logical processors, so it should be named either `get_num_logical_processor` or `num_logical_processor`.  Otherwise, we'd lose the most important concept "number".

Reply:
Got it, thanks, I will use `num_logical_processor` instead.